opacus/accountants/utils.py

```diff
@@ -39,27 +38,31 @@ def get_noise_multiplier(
         raise NotImplementedError(
             "get_noise_multiplier is currently only supports RDP accountant"
         )
+    if (steps is None) == (epochs is None):
+        raise ValueError(
+            "get_noise_multiplier takes as input EITHER a number of steps or a number of epochs"
+        )
+    if steps is None:
+        steps = int(epochs / sample_rate)
 
     eps = float("inf")
-    sigma_min = DEFAULT_SIGMA_MIN_BOUND
-    sigma_max = DEFAULT_SIGMA_MAX_BOUND
     accountant = create_accountant(mechanism=accountant)
 
     while eps > target_epsilon:
-        sigma_max = 2 * sigma_max
-        accountant.steps = [(sigma_max, sample_rate, int(epochs / sample_rate))]
+        sigma_high = 2 * sigma_high
+        accountant.steps = [(sigma_high, sample_rate, steps)]
         eps = accountant.get_epsilon(delta=target_delta, **kwargs)
-        if sigma_max > MAX_SIGMA:
+        if sigma_high > sigma_max:
             raise ValueError("The privacy budget is too low.")
 
-    while sigma_max - sigma_min > SIGMA_PRECISION:
-        sigma = (sigma_min + sigma_max) / 2
-        accountant.steps = [(sigma, sample_rate, int(epochs / sample_rate))]
+    while sigma_high - sigma_low > sigma_precision:
+        sigma = (sigma_low + sigma_high) / 2
+        accountant.steps = [(sigma, sample_rate, steps)]
         eps = accountant.get_epsilon(delta=target_delta, **kwargs)
 
         if eps < target_epsilon:
-            sigma_max = sigma
+            sigma_high = sigma
         else:
-            sigma_min = sigma
+            sigma_low = sigma
```

Comment:
I'm thinking if we can use [`scipy.optimzie.bisect()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.bisect.html) instead of implementing our own.
It is easy to implement binary search incorrectly so we need to make tests for all corner cases. If we use scipy, we don't need to add our own tests, plus it is much more stable than the rest of our dependencies.